src/v2/coins/abstractUtxoCoin.js

```diff
@@ -1071,16 +1074,24 @@ class AbstractUtxoCoin extends BaseCoin {
         return walletUnspents;
```

Comment:
It's a bit strange that you close over `walletUnspents` instead of passing it in as a param (or returning an array instead of appending to the closed over variable). But it should work. (wanted to leave this comment on line 1051, but github won't let me)

Reply:
Ah yeah this is some funky logic. I didn't write this part. It was Arik haha. Happy to refactor if you think its worth it